airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -280,6 +277,148 @@ def test_secret_is_properly_marked(self, connector_spec_dict: dict, detailed_log
                     {properties}"""
             )
 
+    def test_property_type_is_not_array(self, connector_spec_dict: dict):
+        """
+        Each field has one or multiple types, but the UI only supports a single type and optionally "null" as a second type.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
```

Comment:
`connectionSpecification` could be a class fixture as you reuse it quite often. Scoping it to `class` will avoid reloading it for each test.

```suggestion
    def test_property_type_is_not_array(self, connection_specification: dict):
        """
        Each field has one or multiple types, but the UI only supports a single type and optionally "null" as a second type.
        """
```

You can declare it in the class with:
```python
@pytest.fixture(scope="class")
def connection_specification(self, connector_spec_dict):
    return connector_spec_dict["connectionSpecification"]
```